sql-api/src/main/java/io/cdap/pipeline/sql/api/template/TableInfo.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.pipeline.sql.api.template;
+
+import com.google.common.collect.ImmutableList;
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.rel.type.RelDataTypeFactory;
+import org.apache.calcite.rel.type.StructKind;
+import org.apache.calcite.schema.impl.AbstractTable;
+import org.apache.calcite.sql.type.SqlTypeName;
+
+import java.util.ArrayList;
+import java.util.List;
+import javax.annotation.Nullable;
+
+/**
+ * Represents a Calcite Table with a schema and no rows.
+ */
+public class TableInfo extends AbstractTable {
+  String tableName;
+  List<String> columnNames;
+  List<SqlTypeName> columnTypes;
+  RelDataType type;
+
+  private TableInfo(String tableName, List<String> columnNames,
+                    List<SqlTypeName> columnTypes, @Nullable RelDataType type) {
+    this.tableName = tableName;
+    this.columnNames = ImmutableList.copyOf(columnNames);
```

Comment:
It's better not to have guava in the api, because of compatibility issues. We would like to not force plugins to use a specific guava version. Is it coming from calcite-core?

In any case, it's better to use plain java and do `Collections.unmodifiableList(new ArrayList<>(columnNames))` here, in case Calcite ever removes it.

Reply:
Fixed